src/flow.ts

```diff
@@ -1,5 +1,27 @@
 import { gql } from "graphql-request";
 import type { GraphQLClient } from "graphql-request";
+import type { FlowGraph } from "./types";
+
+export async function getLatestFlowGraph(
+  client: GraphQLClient,
+  flowId: string
+): Promise<FlowGraph | undefined> {
+  const { published_flows: response } = await client.request(
+    gql`
+      query GetLatestPublishedFlowData($flowId: uuid!) {
+        published_flows(
+          where: { flow_id: { _eq: $flowId } }
+          order_by: { created_at: desc }
+          limit: 1
+        ) {
+          data
+        }
+      }
+    `,
+    { flowId }
+  );
+  return response?.length ? response[0].data : null;
```

Comment:
nit: can this `return response?.[0]?.data`? otherwise i think falling back to `null` fails to  align with the the function type says it'll return `Promise<... | undefined>`?

Reply:
Weird, I would have expected Typescript to have spotted this :shrug: 
Thanks, I've fixed the type to return null as intended.